### ISSUE TITLE
Repo: Replace generate:rules-lists script with breaking changes generator

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5928]
### Suggestion

We still have a `generate:rules-lists` script in `packages/eslint-plugin/package.json`. But it doesn't work because there haven't been manual Markdown tables of rules checked into the repo for quite some time (#5096).

In parallel, for breaking changes to the recommended rulesets (#5900), it'd be nice to have that table generated automatically.

Out with the old, in with the new!
